test/metabase/api/search_test.clj

```diff
@@ -379,7 +379,14 @@
                                  :name     "test segment"}]]
       (perms/revoke-data-perms! (perms-group/all-users) db-id)
       (is (= []
-             (search-request-data :rasta :q "test"))))))
+             (search-request-data :rasta :q "test")))))
+
+  (testing "Databases for which the user does not have access to should not show up in results"
+    (mt/with-temp* [Database       [db-1 {:name "db-1"}]
+                    Database       [_db-2 {:name "db-2"}]]
+      (perms/revoke-data-perms! (perms-group/all-users) (:id db-1))
```

Comment:
nit: I'd add another test to check if the requests return both `db-1` and `db-2` before the revoke perms.

Reply:
It doesn't hurt to add another test, so I've added another `is` expression before revoking perms https://github.com/metabase/metabase/pull/23678/commits/ca0a349cd8971d79237c61dd28558ab393d48d11